File: 38-count-and-say/count-and-say.java
class Solution {
    String check(int i,String s,int n){
        if(i==n) return s;
        int count=0,ni=s.length();
        char pre=' ';
        String ans="";

        for(int j=0;j<ni;j++){
            char c=s.charAt(j);
            if(c!=pre){
                if(count!=0){
                    String cc=Integer.toString(count);
                    ans=ans+cc+pre;
                }
                pre=c;
                count=1;
            }
            else count++;
        }
        String cc=Integer.toString(count);
        ans=ans+cc+pre;

        return check(i+1,ans,n);
    }
    public String countAndSay(int n) {
        return check(1,"1",n);
    }
}

File: 38-count-and-say/count-and-say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """

        def check(no):
            if no == 1:
                return "1"
            
            num = check(no - 1)
            ans = ""
            l = 0 
            r = 1
            # print(no, num)
            while r < len(num):
                if num[r] == num[r - 1]:
                    r += 1
                    continue
                ans += str((r - l)) + num[l]
                l = r
                r += 1
                
            ans += str((r - l)) + num[l]
            print(no, num, ans)
            return ans

        k = check(n)
        return k
        